server/src-lib/Hasura/GraphQL/Resolve/Action.hs

```diff
@@ -414,10 +420,10 @@ callWebhook manager outputType outputFields reqHeaders confHeaders
                pure (webhookResponse, mkSetCookieHeaders responseWreq)
 
          | HTTP.statusIsClientError responseStatus -> do
-             ActionWebhookErrorResponse message maybeCode <-
+             ActionWebhookErrorResponse message maybeExtension <-
                modifyErr ("webhook response: " <>) $ decodeValue responseValue
-             let code = maybe Unexpected ActionWebhookCode maybeCode
-                 qErr = QErr [] responseStatus message code Nothing
+             let qErr = QErr [] responseStatus message Unexpected $
+                   (maybe Nothing (Just . J.toJSON . ActionWebhookErrorExtension) maybeExtension)
              throwError qErr
```

Comment:
`qeInternal` of `QErr` will go into `extensions` of the response json. So the response look like
```
extensions:
  internal:
    extensions: response-json-from-webhook
```
See `encodeGQLErr ` function. 

I'd suggest to model the `extensions` in the `QErr` type.

```diff
+data ErrorExtra
+  = EEInternal !Value
+  | EEExtensions !Value
+  deriving (Show, Eq)
+
+instance ToJSON ErrorExtra where
+  toJSON = \case
+    EEInternal v -> v
+    EEExtensions v -> v
+
 data QErr
   = QErr
-  { qePath     :: !JSONPath
-  , qeStatus   :: !N.Status
-  , qeError    :: !T.Text
-  , qeCode     :: !Code
-  , qeInternal :: !(Maybe Value)
+  { qePath   :: !JSONPath
+  , qeStatus :: !N.Status
+  , qeError  :: !T.Text
+  , qeCode   :: !Code
+  , qeExtra  :: !(Maybe ErrorExtra)
   } deriving (Show, Eq)
 
 instance ToJSON QErr where
@@ -157,12 +167,12 @@ instance ToJSON QErr where
     , "error" .= msg
     , "code"  .= show code
     ]
-  toJSON (QErr jPath _ msg code (Just ie)) =
+  toJSON (QErr jPath _ msg code (Just extra)) =
     object
     [ "path"  .= encodeJSONPath jPath
     , "error" .= msg
     , "code"  .= show code
-    , "internal" .= ie
+    , "internal" .= extra
     ]
 
 noInternalQErrEnc :: QErr -> Value
@@ -174,18 +184,20 @@ noInternalQErrEnc (QErr jPath _ msg code _) =
   ]
 
 encodeGQLErr :: Bool -> QErr -> Value
-encodeGQLErr includeInternal (QErr jPath _ msg code mIE) =
+encodeGQLErr includeInternal (QErr jPath _ msg code mExtra) =
   object
   [ "message" .= msg
   , "extensions" .= extnsObj
   ]
   where
-    extnsObj = object $ bool codeAndPath
-               (codeAndPath ++ internal) includeInternal
+    extnsObj = case mExtra of
+      Nothing -> object codeAndPath
+      Just (EEExtensions v) -> v
+      Just (EEInternal v) -> object $
+        bool codeAndPath (codeAndPath ++ ["internal" .= v]) includeInternal
     codeAndPath = [ "code" .= show code
                   , "path" .= encodeJSONPath jPath
                   ]
-    internal = maybe [] (\ie -> ["internal" .= ie]) mIE
```